frontend/packages/console-plugin-sdk/src/typings/pvc.ts

```diff
@@ -0,0 +1,51 @@
+import { K8sResourceCommon } from '@console/internal/module/k8s';
+import { Extension, LazyLoader } from './base';
+
+namespace ExtensionProperties {
+  export interface PVCAlert<T> {
+    /** Loader for the corresponding Alert component. */
+    loader: LazyLoader<T>;
+  }
+
+  export interface PVCStatus<T> {
+    /** Loader for the corresponding Status component. */
+    loader: LazyLoader<T>;
+    /** predicate that tells if to render the status or not */
+    predicate: (pvc: K8sResourceCommon) => boolean;
+    /** priority for the corresponding Status component. bigger is prioritized */
+    priority: number;
+  }
+
+  export interface PVCCreateProp {
+    /** key for the create prop */
+    key: string;
```

Comment:
do you actually need to specify `key` ? What if there are two extensions with same key ? I think you should just generate the `key` in code, using index is fine I think.